scripts/dts/gen_defines.py

```diff
@@ -82,6 +106,9 @@ def main():
             write_bus(node)
             write_existence_flags(node)
 
+            out_dev(node, "ZDID", node_to_zdid[node])
```

Comment:
Do we need to generate `..._ZDID` macros for all the device aliases and stuff too, like `out_dev()` does? If not, maybe the ID could be special-cased and just generate the needed macro.

The format of the macro could be modified then too, so that it doesn't clash with regular properties.

What do you think about having

```
DT_DEV_ID_INST_##inst##_##dev_name##
```

instead of

```
DT_INST_##inst##_##dev_name##_ZDID
```

Something more plain-English and guessable like "DEV_ID" seems nicer than "ZDID". Already know we're dealing with Zephyr. ;)